compiler/qsc_ast/src/ast.rs

```diff
@@ -1306,6 +1306,119 @@ pub struct Ident {
     pub name: Rc<str>,
 }
 
+/// A [`VecIdent`] represents a sequence of idents. It provides a helpful abstraction
+/// that is more powerful than a simple `Vec<Ident>`, and is primarily used to represent
+/// dot-separated paths.
```

Comment:
If perf continues to be a concern, there may be other structures we can look at. For example, VS Code uses implementations of `src/vs/base/common/ternarySearchTree.ts` and `src/vs/base/common/prefixTree.ts` to optimize some lookups (though they are a real pain when debugging)